bin/load-module-esm-support.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+var fs = require('fs');
+var extnamePath = require('path').extname;
+var resolvePath = require('path').resolve;
+
+var cwd = process.cwd();
+var pkgPath = resolvePath(cwd, 'package.json');
+
+var isModulePackage = fs.existsSync(pkgPath) && require(pkgPath).type === 'module';
```

Comment:
package.json files with `type: module` can be nested, and in particular, a package.json of `type: commonjs` can undo this, so the check has to always be relative to the `file`, and has to go all the way up the filesystem until it finds a `type` property.

Reply:
will a nested package.json without a `type` "undo" a parent package.json with `type: module`?

Reply:
No, only if it has an explicit type commonjs

Reply:
i created a basic proof-of-conecpt implementation of traversing up the file tree to find a package.json with a type field (still need to verify this works in windows). i refrained from using for example [find-up](https://github.com/sindresorhus/find-up) since it uses too modern syntax. where should i place tests for this? next to the other, regular tape tests?